test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -710,33 +739,59 @@ static void Enlist(bool durable, bool shouldRollBack, ConcurrentQueue<Transactio
 
             public void Prepare(PreparingEnlistment preparingEnlistment)
             {
-                _eventQueue?.Enqueue(new TransactionEvent(_name + ": prepare phase"));
+                _eventQueue?.Enqueue(new TransactionEvent(_name + ": prepare phase start"));
+                Thread.Sleep(1);
                 if (_shouldRollBack)
                 {
+                    _eventQueue?.Enqueue(new TransactionEvent(_name + ": prepare phase, calling rollback-ed"));
                     preparingEnlistment.ForceRollback();
                 }
                 else
                 {
+                    _eventQueue?.Enqueue(new TransactionEvent(_name + ": prepare phase, calling prepared"));
                     preparingEnlistment.Prepared();
                 }
+                Thread.Sleep(1);
+                _eventQueue?.Enqueue(new TransactionEvent(_name + ": prepare phase end"));
```

Comment:
More logs in order to detect more easily thing occurring concurrently. (Adding the thread number is an indication too, but who knows if those threads are launched concurrently or one after another with some synchronization mechanism?)